src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java

```diff
@@ -333,14 +336,19 @@ public Collection<Object> createComponents(
         this.clientUtil = new ClientUtil(settings, client, throttler, threadPool);
         this.indexUtils = new IndexUtils(client, clientUtil, clusterService, indexNameExpressionResolver);
         this.nodeFilter = new DiscoveryNodeFilterer(clusterService);
-        this.anomalyDetectionIndices = new AnomalyDetectionIndices(
-            client,
-            clusterService,
-            threadPool,
-            settings,
-            nodeFilter,
-            AnomalyDetectorSettings.MAX_UPDATE_RETRY_TIMES
-        );
+        // convert from checked IOException to unchecked RuntimeException
+        this.anomalyDetectionIndices = ThrowingSupplierWrapper
```

Comment:
I am guessing that the supplier would be reused for forecast. Did you want to rename "anomalyDetectionIndices"?

Reply:
I will have a different one for forecast. Their functionalities are different.